src/simulator/ackermann_model.cpp

```diff
@@ -0,0 +1,96 @@
+#include "simulator/ackermann_model.h"
+#include "config_reader/config_reader.h"
+#include "shared/util/timer.h"
+#include "shared/math/math_util.h"
+
+using std::string;
+using f1tenth_simulator::AckermannCurvatureDriveMsg;
+using math_util::AngleDiff;
+using math_util::AngleMod;
+
+namespace ackermann {
+
+CONFIG_FLOAT(cMinTurnR, "min_turn_radius");
+CONFIG_FLOAT(cMaxAccel, "max_accel");
+CONFIG_FLOAT(cMaxSpeed, "max_speed");
+CONFIG_FLOAT(cAngularErrorBias, "angular_error_bias");
+CONFIG_FLOAT(cAngularErrorRate, "angular_error_rate");
+CONFIG_STRING(cDriveTopic, "drive_callback_topic");
+
+
+AckermannModel::AckermannModel(const string& config_file, ros::NodeHandle n) :
+    angular_error_(0, 1) {
+  config_reader::ConfigReader({config_file});
+  // Use the config reader to initialize the subscriber
+  drive_subscriber_ = n.subscribe(
+      cDriveTopic,
+      1,
+      &AckermannModel::DriveCallback,
+      this);
+  template_lines_ = {};
+  pose_lines_ = {};
+}
+
+void AckermannModel::DriveCallback(const AckermannCurvatureDriveMsg& msg) {
+  if (!isfinite(msg.velocity) || !isfinite(msg.curvature)) {
+    printf("Ignoring non-finite drive values: %f %f\n",
+        msg.velocity,
+        msg.curvature);
+    return;
+  }
+  last_cmd_ = msg;
+  t_last_cmd_ = GetMonotonicTime();
+}
+
+float AbsBound(float x, float bound) {
+  if (x > 0.0 && x > bound) {
+    return bound;
+  } else if (x < 0.0 && x < -bound) {
+    return -bound;
+  }
+  return x;
+}
+
+void AckermannModel::Step(const double &dt) {
+  static const double kMaxCommandAge = 0.1;
+  if (GetMonotonicTime() > t_last_cmd_ + kMaxCommandAge) {
```

Comment:
In service of making this support faster than realtime, you should consider a more robust way to count timeouts that's dependent on ticks or some other wallclock time invariant manner.